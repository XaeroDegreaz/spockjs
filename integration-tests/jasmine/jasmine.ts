import { sync as run } from 'cross-spawn';
import { unlinkSync as del, writeFileSync as write } from 'fs';
import { resolve } from 'path';
import { env } from 'process';

import { babelJitEnv, modulePath, typescriptJitEnv } from '../utils';

export default (cwd: string) => {
  const config = resolve(cwd, 'jasmine.json');

  return {
    writeConfig: () =>
      write(
        config,
        JSON.stringify({
          spec_dir: '.',
          spec_files: ['jasmine.js'],
          helpers: [
            resolve(modulePath('@babel/register'), 'lib', 'node.js'),
            resolve(modulePath('ts-node'), 'register', 'index.js'),
          ],
          random: false,
        }),
      ),

    deleteConfig: () => del(config),

    run: () =>
      run('node', [resolve(modulePath('jasmine'), 'bin', 'jasmine')], {
        cwd,
        env: {
          ...env,
          ...babelJitEnv,
          ...typescriptJitEnv,
          JASMINE_CONFIG_PATH: 'jasmine.json',
        },
      }),
  };
};
