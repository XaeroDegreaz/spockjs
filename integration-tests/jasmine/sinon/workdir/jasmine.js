import * as sinon from 'sinon';

describe('jasmine interaction blocks', () => {
  it('stubbing', () => {
    const s = sinon.stub();
    stub: s() >> 42;
    expect(s()).toBe(42);
  });

  it('mocking', () => {
    const obj = { method: () => {} };

    const mock = sinon.mock(obj);
    mock: 2 * mock.method();

    obj.method();
    obj.method();

    verify: mock;
  });

  it('failing verification', () => {
    const obj = { method: () => {} };

    const mock = sinon.mock(obj);
    mock: 2 * mock.method();

    obj.method();

    verify: mock;
  });
});
