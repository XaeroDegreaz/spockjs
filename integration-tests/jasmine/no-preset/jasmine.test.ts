import { resolve } from 'path';

import jasmine from '../jasmine';

// mark implicit dependencies for jest
() => require('./workdir/jasmine.js') && require('./workdir/package.json');

const cwd = resolve(__dirname, 'workdir');

const helper = jasmine(cwd);

beforeAll(helper.writeConfig);
afterAll(helper.deleteConfig);

test('produces correct output', () => {
  const { status, stdout } = helper.run();

  expect: status === 1;
  expect(stdout.toString().replace(/Stack:.*/s, '')).toMatchSnapshot();
});
