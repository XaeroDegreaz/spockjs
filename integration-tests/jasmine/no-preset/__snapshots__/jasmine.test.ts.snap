// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`produces correct output 1`] = `
"Started
.F

Failures:
1) jasmine with spock failing test
  Message:
    AssertionError [ERR_ASSERTION]: (x === double(2)) is not truthy   # jasmine.js:11
      
      _assert(x === double(2), \\"(x === double(2)) is not truthy\\")
              | |   |                                            
              | |   4                                            
              3 false                                            
      
      [number] double(2)
      => 4
      [number] x
      => 3
      
  "
`;
